web-registry/src/lib/wallet/hooks/useLogin.tsx

```diff
@@ -0,0 +1,55 @@
+import { useCallback } from 'react';
+import { useQuery } from '@tanstack/react-query';
+
+import { UseStateSetter } from 'types/react/use-state';
+import { getCsrfTokenQuery } from 'lib/queries/react-query/registry-server/getCsrfTokenQuery/getCsrfTokenQuery';
+import { SignArbitraryType, Wallet } from 'lib/wallet/wallet';
+
+import { getAxiosInstance } from '../wallet.utils';
+
+type Params = {
+  signArbitrary?: SignArbitraryType;
+  setError: UseStateSetter<unknown>;
+  setAccountId: UseStateSetter<string | undefined>;
+};
+
+export const useLogin = ({ signArbitrary, setError, setAccountId }: Params) => {
+  // Step 1: Retrieve and save the CSRF tokens
+  const { data: token } = useQuery(getCsrfTokenQuery({}));
+  const login = useCallback(
+    async (wallet?: Wallet): Promise<void> => {
+      try {
+        if (wallet?.address && signArbitrary && token) {
+          const instance = getAxiosInstance(token);
+
+          // Step 2: Retrieve a nonce for the user
+          const nonceRes = await instance.get('web3auth/nonce', {
+            params: { userAddress: wallet.address },
+            validateStatus: (status: number) => {
+              // 404 error might be returned in case of new user,
```

Comment:
one thought i had is that perhaps a tracking event for new user account created could be useful here (when status=404)

Reply:
I'll merge this for now, let's create an issue to update/create tracking events if needed